lib/spack/spack/test/conftest.py

```diff
@@ -980,6 +983,38 @@ def install_mockery(temporary_store: spack.store.Store, mutable_config, mock_pac
     temporary_store.failure_tracker.clear_all()
 
 
+@pytest.fixture(scope="module")
+def temporary_mirror_dir(tmpdir_factory):
+    dir = tmpdir_factory.mktemp("mirror")
+    dir.ensure("build_cache", dir=True)
+    yield str(dir)
+    dir.join("build_cache").remove()
+
+
+@pytest.fixture(scope="function")
+def temporary_mirror(temporary_mirror_dir):
+    mirror_url = url_util.path_to_file_url(temporary_mirror_dir)
+    mirror_cmd("add", "--scope", "site", "test-mirror-func", mirror_url)
+    yield temporary_mirror_dir
+    mirror_cmd("rm", "--scope=site", "test-mirror-func")
```

Comment:
In addition to adding the import around line 46:

```suggestion
    mirror_name = "test-mirror-func"
    mirror = spack.mirror.Mirror(mirror_url, name=mirror_name)
    spack.mirror.add(mirror, "site")
    yield temporary_mirror_dir
    spack.mirror.remove(mirror_name, "site")
```

🤔 Part of me is wondering if this should be within a config override context manager but I've only run this by one of the unit test so far.